### PR TITLE
Run common tests for ReplayFileSystem.

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

**Comment by tvolkert** [comment]
@cbracken

File: test/common_tests.dart [hunk 1]
```diff
@@ -10,12 +10,20 @@ import 'dart:io' as io;
...
 /// file system entities will be created.
 typedef String RootPathGenerator();
 
+/// Callback used in [runCommonTests] to create the file system under test.
+/// It must return either a [FileSystem] or a [Future] that completes with a
+/// [FileSystem].
+typedef dynamic FileSystemGenerator();
+
+/// A function to run before tests (passed to [setUp]).
+typedef dynamic SetUpCallback();
```
Comment (cbracken) [inline test/common_tests.dart]: Not a huge fan of `Callback` in the type, but I can't think of a better name; `SetUpFunction` feels a bit more Dart-y to me. Out of desperation, I looked at `package:test` and they use `AsyncFunction`, though they call it `callback` in the formal parameter list for `setUp()`. 

TL;DR unless you can think of a name that's significantly better I don't think there's anything to do here; carry on :)

**Comment by cbracken** [comment]
lgtm
